### DISCUSSION TITLE
Lock question in DBeaver 21.2.4 - Metadata

### DISCUSSION THREAD

**Discussion by JCUtah** [dbeaver/dbeaver#14513] [Q&A]
#### System information: 
- System Windows 10 Pro
- DBeaver 21.2.2
- Java 11.0.12+7 
- Additional extensions - none

#### Connection specification:
- Database name and version - PostgreSQL 13.2 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44), 64-bit running on Centos 7
- Driver name - PostgreSQL JDBC Driver
- Do you use tunnels or proxies (SSH, SOCKS, etc)? - No

#### Describe the problem you're observing:
I have a procedure that calls another procedure that builds 4 tables. It's going into a lock state occasionally when it runs and bumps against pg_dump. The application column of the lock query says DBeaver 21.2.4 - Metadata <dbname> and the query is:

select c.oid,pg_catalog.pg_total_relation_size(c.oid) as total_rel_size,pg_catalog.pg_relation_size(c.oid) as rel_size
FROM pg_class c
WHERE c.relnamespace=$1

The query isn't in my procedure so I'm assuming that DBeaver is executing it to get metadata information although it may PostgreSQL itself (you know what they say about assumptions :) ) 
Can anyone explain to me what the query is for and why it might run for hours?

#### Include any warning/errors/backtraces from the logs
None

**Comment by LonwoLonwo** [comment]
Hello @JCUtah 
DBeaver uses this query for tables statistics reading.

You can turn off reading statistics in the navigator settings:

![2021-11-09 10_25_21-DBeaver Enterprise 21 2 4 - _postgres12local_ Script](https://user-images.githubusercontent.com/45152336/140880332-9a903b03-80a9-4768-9f54-89ddca8710b5.png)

**Comment by JCUtah** [comment]
Thank you for the quick response.
